src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetails.cs

```diff
@@ -25,9 +28,46 @@ public abstract class RethrowToPreserveStackDetailsAnalyzer : DiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
-        protected static Diagnostic CreateDiagnostic(SyntaxNode node)
+        public override void Initialize(AnalysisContext analysisContext)
         {
-            return node.CreateDiagnostic(Rule);
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(context =>
+            {
+                var throwOperation = (IThrowOperation)context.Operation;
+
+                if (throwOperation.Exception is not ILocalReferenceOperation localReference)
+                {
+                    return;
+                }
+
+                IOperation ancestor = throwOperation;
+                while (ancestor != null &&
+                    ancestor.Kind != OperationKind.AnonymousFunction &&
+                    ancestor.Kind != OperationKind.LocalFunction)
```

Comment:
OperationKind.MethodBody ?

Reply:
I'm not sure what OperationKind.FlowAnonymousFunction. But assuming it's some special kind of AnonymousFunction, you might want to consider it.

Reply:
I don't think you need to handle MethodBody, that will anyway be the root operation.

Reply:
> I don't think you need to handle MethodBody, that will anyway be the root operation.

Yes, exactly! 

> I'm not sure what OperationKind.FlowAnonymousFunction. But assuming it's some special kind of AnonymousFunction, you might want to consider it.

Looking at the [doc](https://docs.microsoft.com/dotnet/api/microsoft.codeanalysis.flowanalysis.iflowanonymousfunctionoperation) it seems fine to ignore.